--- a/drivers/net/wireless/mwifiex/pcie.c
+++ b/drivers/net/wireless/mwifiex/pcie.c
@@ -271,9 +271,14 @@ static struct pci_driver __refdata mwifi
 	.probe    = mwifiex_pcie_probe,
 	.remove   = mwifiex_pcie_remove,
 #ifdef CONFIG_PM_SLEEP
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29))
 	.driver   = {
 		.pm = &mwifiex_pcie_pm_ops,
 	},
+#else
+	.suspend    = mwifiex_pcie_suspend_compat,
+	.resume     = mwifiex_pcie_resume_compat,
+#endif
 #endif
 	.shutdown = mwifiex_pcie_shutdown,
 };
