--- a/drivers/net/wireless/ath/ath6kl/main.c
+++ b/drivers/net/wireless/ath/ath6kl/main.c
@@ -1208,7 +1208,11 @@ static void ath6kl_set_multicast_list(st
 	list_for_each_entry_safe(mc_filter, tmp, &vif->mc_filter, list) {
 		found = false;
 		netdev_for_each_mc_addr(ha, ndev) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 			if (memcmp(ha->addr, mc_filter->hw_addr,
+#else
+			if (memcmp(ha->dmi_addr, mc_filter->hw_addr,
+#endif
 				   ATH6KL_MCAST_FILTER_MAC_ADDR_SIZE) == 0) {
 				found = true;
 				break;
@@ -1242,7 +1246,11 @@ static void ath6kl_set_multicast_list(st
 	netdev_for_each_mc_addr(ha, ndev) {
 		found = false;
 		list_for_each_entry(mc_filter, &vif->mc_filter, list) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 			if (memcmp(ha->addr, mc_filter->hw_addr,
+#else
+			if (memcmp(ha->dmi_addr, mc_filter->hw_addr,
+#endif
 				   ATH6KL_MCAST_FILTER_MAC_ADDR_SIZE) == 0) {
 				found = true;
 				break;
@@ -1257,7 +1265,11 @@ static void ath6kl_set_multicast_list(st
 				goto out;
 			}
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 			memcpy(mc_filter->hw_addr, ha->addr,
+#else
+			memcpy(mc_filter->hw_addr, ha->dmi_addr,
+#endif
 			       ATH6KL_MCAST_FILTER_MAC_ADDR_SIZE);
 			/* Set the multicast filter */
 			ath6kl_dbg(ATH6KL_DBG_TRC,
