--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -823,8 +823,13 @@ static void ieee80211_do_stop(struct iee
 	if (sdata->dev) {
 		netif_addr_lock_bh(sdata->dev);
 		spin_lock_bh(&local->filter_lock);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 		__hw_addr_unsync(&local->mc_list, &sdata->dev->mc,
 				 sdata->dev->addr_len);
+#else
+		__dev_addr_unsync(&local->mc_list, &local->mc_count,
+				  &sdata->dev->mc_list, &sdata->dev->mc_count);
+#endif
 		spin_unlock_bh(&local->filter_lock);
 		netif_addr_unlock_bh(sdata->dev);
 	}
@@ -1024,7 +1029,12 @@ static void ieee80211_set_multicast_list
 		sdata->flags ^= IEEE80211_SDATA_PROMISC;
 	}
 	spin_lock_bh(&local->filter_lock);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 	__hw_addr_sync(&local->mc_list, &dev->mc, dev->addr_len);
+#else
+	__dev_addr_sync(&local->mc_list, &local->mc_count,
+			&dev->mc_list, &dev->mc_count);
+#endif
 	spin_unlock_bh(&local->filter_lock);
 	ieee80211_queue_work(&local->hw, &local->reconfig_filter);
 }
