--- a/drivers/net/ethernet/atheros/alx/main.c
+++ b/drivers/net/ethernet/atheros/alx/main.c
@@ -1264,8 +1264,10 @@ static int alx_probe(struct pci_dev *pde
 	netdev->irq = pdev->irq;
 	netdev->watchdog_timeo = ALX_WATCHDOG_TIME;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24)
 	if (ent->driver_data & ALX_DEV_QUIRK_MSI_INTX_DISABLE_BUG)
 		pdev->dev_flags |= PCI_DEV_FLAGS_MSI_INTX_DISABLE_BUG;
+#endif
 
 	err = alx_init_sw(alx);
 	if (err) {
