--- a/drivers/net/ethernet/atheros/atlx/atlx.c
+++ b/drivers/net/ethernet/atheros/atlx/atlx.c
@@ -250,6 +250,7 @@ static void atlx_restore_vlan(struct atl
 	atlx_vlan_mode(adapter->netdev, adapter->netdev->features);
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 static netdev_features_t atlx_fix_features(struct net_device *netdev,
 	netdev_features_t features)
 {
@@ -275,5 +276,6 @@ static int atlx_set_features(struct net_
 
 	return 0;
 }
+#endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39)) */
 
 #endif /* ATLX_C */
