--- a/drivers/net/wireless/ath/ath6kl/main.c
+++ b/drivers/net/wireless/ath/ath6kl/main.c
@@ -1119,6 +1119,7 @@ static struct net_device_stats *ath6kl_g
 	return &vif->net_stats;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 static int ath6kl_set_features(struct net_device *dev,
 			       netdev_features_t features)
 {
@@ -1151,6 +1152,7 @@ static int ath6kl_set_features(struct ne
 
 	return err;
 }
+#endif
 
 static void ath6kl_set_multicast_list(struct net_device *ndev)
 {
@@ -1298,7 +1300,9 @@ static const struct net_device_ops ath6k
 	.ndo_stop               = ath6kl_close,
 	.ndo_start_xmit         = ath6kl_data_tx,
 	.ndo_get_stats          = ath6kl_get_stats,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 	.ndo_set_features       = ath6kl_set_features,
+#endif
 	.ndo_set_rx_mode	= ath6kl_set_multicast_list,
 };
 
@@ -1315,7 +1319,11 @@ void init_netdev(struct net_device *dev)
 					WMI_MAX_TX_META_SZ +
 					ATH6KL_HTC_ALIGN_BYTES, 4);
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 	dev->hw_features |= NETIF_F_IP_CSUM | NETIF_F_RXCSUM;
+#else
+	dev->features |= NETIF_F_IP_CSUM;
+#endif
 
 	return;
 }
