--- a/drivers/net/wireless/orinoco/hermes.c
+++ b/drivers/net/wireless/orinoco/hermes.c
@@ -766,7 +766,7 @@ static const struct hermes_ops hermes_op
 	.write_ltv = hermes_write_ltv,
 	.bap_pread = hermes_bap_pread,
 	.bap_pwrite = hermes_bap_pwrite,
-	.read_pda = hermes_read_pda,
+	.read_pda_h = hermes_read_pda,
 	.program_init = hermesi_program_init,
 	.program_end = hermesi_program_end,
 	.program = hermes_program_bytes,
--- a/drivers/net/wireless/orinoco/hermes.h
+++ b/drivers/net/wireless/orinoco/hermes.h
@@ -393,7 +393,7 @@ struct hermes_ops {
 			 u16 id, u16 offset);
 	int (*bap_pwrite)(struct hermes *hw, int bap, const void *buf,
 			  int len, u16 id, u16 offset);
-	int (*read_pda)(struct hermes *hw, __le16 *pda,
+	int (*read_pda_h)(struct hermes *hw, __le16 *pda,
 			u32 pda_addr, u16 pda_len);
 	int (*program_init)(struct hermes *hw, u32 entry_point);
 	int (*program_end)(struct hermes *hw);
