--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -1990,6 +1990,7 @@ netdev_tx_t ieee80211_subif_start_xmit(s
 		goto fail_rcu;
 	}
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0))
 	if (unlikely(!multicast && skb->sk &&
 		     skb_shinfo(skb)->tx_flags & SKBTX_WIFI_STATUS)) {
 		struct sk_buff *orig_skb = skb;
@@ -2018,6 +2019,7 @@ netdev_tx_t ieee80211_subif_start_xmit(s
 			skb = orig_skb;
 		}
 	}
+#endif
 
 	/*
 	 * If the skb is shared we need to obtain our own copy.
