--- a/drivers/gpu/drm/drm_sysfs.c
+++ b/drivers/gpu/drm/drm_sysfs.c
@@ -98,7 +98,11 @@ static const struct dev_pm_ops drm_class
 	.freeze		= drm_class_freeze,
 };
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0))
 static char *drm_devnode(struct device *dev, umode_t *mode)
+#else
+static char *drm_devnode(struct device *dev, mode_t *mode)
+#endif
 {
 	return kasprintf(GFP_KERNEL, "dri/%s", dev_name(dev));
 }
