--- a/net/bluetooth/sco.c
+++ b/net/bluetooth/sco.c
@@ -430,8 +430,12 @@ static struct sock *sco_sock_alloc(struc
 	return sk;
 }
 
+#if defined(CPTCFG_BACKPORT_OPTION_BT_SOCK_CREATE_NEEDS_KERN)
 static int sco_sock_create(struct net *net, struct socket *sock, int protocol,
 			   int kern)
+#else
+static int sco_sock_create(struct net *net, struct socket *sock, int protocol)
+#endif
 {
 	struct sock *sk;
 
@@ -726,7 +730,11 @@ static int sco_sock_recvmsg(struct kiocb
 	return bt_sock_recvmsg(iocb, sock, msg, len, flags);
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,31))
 static int sco_sock_setsockopt(struct socket *sock, int level, int optname, char __user *optval, unsigned int optlen)
+#else
+static int sco_sock_setsockopt(struct socket *sock, int level, int optname, char __user *optval, int optlen)
+#endif
 {
 	struct sock *sk = sock->sk;
 	int len, err = 0;
