--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -859,6 +859,9 @@ static int i915_gmch_probe(struct drm_de
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	int ret;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(3,9,0))
+	const struct intel_gtt *gtt;
+#endif
 
 	ret = intel_gmch_probe(dev_priv->bridge_dev, dev_priv->dev->pdev, NULL);
 	if (!ret) {
@@ -866,7 +869,16 @@ static int i915_gmch_probe(struct drm_de
 		return -EIO;
 	}
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,9,0))
 	intel_gtt_get(gtt_total, stolen, mappable_base, mappable_end);
+#else
+	gtt = intel_gtt_get();
+
+	*gtt_total = gtt->gtt_total_entries << PAGE_SHIFT;
+	*stolen = gtt->stolen_size;
+	*mappable_base = gtt->gma_bus_addr;
+	*mappable_end = gtt->gtt_mappable_entries << PAGE_SHIFT;
+#endif
 
 	dev_priv->gtt.do_idle_maps = needs_idle_maps(dev_priv->dev);
 	dev_priv->gtt.base.clear_range = i915_ggtt_clear_range;
