--- a/drivers/net/usb/sierra_net.c
+++ b/drivers/net/usb/sierra_net.c
@@ -695,7 +695,7 @@ static int sierra_net_bind(struct usbnet
 
 	priv->usbnet = dev;
 	priv->ifnum = ifacenum;
-	dev->net->netdev_ops = &sierra_net_device_ops;
+	netdev_attach_ops(dev->net, &sierra_net_device_ops);
 
 	/* change MAC addr to include, ifacenum, and to be unique */
 	dev->net->dev_addr[ETH_ALEN-2] = atomic_inc_return(&iface_counter);
