diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index 3cf9600..5e519ce 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -2829,6 +2829,7 @@ static void igb_ethtool_complete(struct
 	pm_runtime_put(&adapter->pdev->dev);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
 static u32 igb_get_rxfh_indir_size(struct net_device *netdev)
 {
 	return IGB_RETA_SIZE;
@@ -2844,6 +2845,7 @@ static int igb_get_rxfh_indir(struct net
 
 	return 0;
 }
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0) */
 
 void igb_write_rss_indir_tbl(struct igb_adapter *adapter)
 {
@@ -2880,6 +2882,7 @@ void igb_write_rss_indir_tbl(struct igb_
 	}
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
 static int igb_set_rxfh_indir(struct net_device *netdev, const u32 *indir)
 {
 	struct igb_adapter *adapter = netdev_priv(netdev);
@@ -2912,6 +2915,7 @@ static int igb_set_rxfh_indir(struct net
 
 	return 0;
 }
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0) */
 
 static unsigned int igb_max_channels(struct igb_adapter *adapter)
 {
@@ -3034,9 +3038,11 @@ static const struct ethtool_ops igb_etht
 	.get_module_info	= igb_get_module_info,
 	.get_module_eeprom	= igb_get_module_eeprom,
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)
 	.get_rxfh_indir_size	= igb_get_rxfh_indir_size,
 	.get_rxfh_indir		= igb_get_rxfh_indir,
 	.set_rxfh_indir		= igb_set_rxfh_indir,
+#endif
 	.get_channels		= igb_get_channels,
 	.set_channels		= igb_set_channels,
 	.begin			= igb_ethtool_begin,
